mmv1/templates/terraform/examples/secure_source_manager_instance_private.tf.erb

```diff
@@ -0,0 +1,73 @@
+resource "google_privateca_ca_pool" "ca_pool" {
+  name     = "<%= ctx[:vars]['ca_pool_id'] %>"
+  location = "us-central1"
+  tier     = "ENTERPRISE"
+  publishing_options {
+    publish_ca_cert = true
+    publish_crl     = true
+  }
+}
+
+resource "google_privateca_certificate_authority" "root_ca" {
+  pool                     = google_privateca_ca_pool.ca_pool.name
+  certificate_authority_id = "<%= ctx[:vars]['root_ca_id'] %>"
+  location                 = "us-central1"
+  deletion_protection = false
+  ignore_active_certificates_on_deletion = true
+  skip_grace_period = true
+  config {
+    subject_config {
+      subject {
+        organization = "google"
+        common_name = "my-certificate-authority"
+      }
+    }
+    x509_config {
+      ca_options {
+        is_ca = true
+      }
+      key_usage {
+        base_key_usage {
+          cert_sign = true
+          crl_sign = true
+        }
+        extended_key_usage {
+          server_auth = true
+        }
+      }
+    }
+  }
+  key_spec {
+    algorithm = "RSA_PKCS1_4096_SHA256"
+  }
+}
+
+resource "google_privateca_ca_pool_iam_binding" "ca_pool_binding" {
+  ca_pool = google_privateca_ca_pool.ca_pool.id
+  role = "roles/privateca.certificateRequester"
+
+  members = [
+    "serviceAccount:service-${data.google_project.project.number}@gcp-sa-sourcemanager.iam.gserviceaccount.com"
+  ]
+}
+
+resource "google_secure_source_manager_instance" "<%= ctx[:primary_resource_id] %>" {
+  instance_id = "<%= ctx[:vars]['instance_id'] %>"
+  location = "us-central1"
+  private_config {
+    is_private = true
+    ca_pool = google_privateca_ca_pool.ca_pool.id
+  }
+  depends_on = [
+    google_privateca_certificate_authority.root_ca,
+    time_sleep.wait_60_seconds
+  ]
+}
+
+resource "time_sleep" "wait_60_seconds" {
+  depends_on = [google_privateca_ca_pool_iam_binding.ca_pool_binding]
+
+  create_duration = "60s"
+}
```

Comment:
Do we need this dependency? It's causing the test to fail since we don't have access to it when executing.

We can add it, but pull providers externally causing test latency (and isn't recommended).
https://github.com/GoogleCloudPlatform/magic-modules/blob/3547ac8d589d09f0b66f4c1fe3e1dfd9783c87d0/mmv1/provider/terraform/examples.rb#L146-L149

Reply:
It seems that the IAM permissions on the privateca_ca_pool take a min or two to propagate, which causes the create instance operation afterwards to fail with insufficient permissions on the service account. Is there a different way to wait a little a while?

Reply:
I think lets just add pull_external if we need to.

We could do 
```
resource "null_resource" "wait_time" {
  provisioner "local-exec" {
    command = "sleep 60"
  }
}
```

But since this is going to be documented externally I would rather avoid it.

Reply:
Ah I will add pull_external then